packages/calypso-e2e/src/lib/pages/general-settings-page.ts

```diff
@@ -0,0 +1,52 @@
+import assert from 'assert';
+import { Page } from 'playwright';
+
+const selectors = {
+	// Common selectors
+	button: ( text: string ) => `button:text-is("${ text }")`,
+
+	// Site Tools
+	deleteSiteLinkItem: `p:text-is("Delete your site permanently")`,
+	deleteSiteConfirmInput: `input[id="confirmDomainChangeInput"]`,
+	deleteSiteConfirmationSpan: 'span:has-text("is being deleted")',
+};
+
+/**
+ * Represents the Settings > General Settings page.
+ */
+export class GeneralSettingsPage {
+	private page: Page;
+
+	/**
+	 * Constructs an instance of the component.
+	 *
+	 * @param {Page} page The underlying page.
+	 */
+	constructor( page: Page ) {
+		this.page = page;
+	}
+
+	/**
+	 * Performs steps related to site deletion.
+	 *
+	 * @param {string} url URL of the site to be deleted.
+	 */
+	async deleteSite( url: string ): Promise< void > {
+		await this.page.click( selectors.deleteSiteLinkItem );
+		await this.page.click( selectors.button( 'Delete site' ) );
+		await this.page.fill( selectors.deleteSiteConfirmInput, url );
+
+		// Once the confirmation button is clicked these things happen in short order:
+		//	- site deletion confirmation toast is shown.
+		//	- navigation to a new page, entering state where no site is selected.
+		await Promise.all( [
+			this.page.waitForNavigation(),
+			this.page.waitForSelector( selectors.deleteSiteConfirmationSpan ),
+			this.page.click( selectors.button( 'Delete this site' ) ),
+		] );
+
+		// Redirected page should no longer reference the deleted site in the URL.
```

Comment:
I recall there was a similar thread of discussion some weeks ago, when you first joined the team. 

I do not recall what the conclusion of that discussion was. However, assertions in page objects get enough searches that I get a good number of results when I query this.

For the most part, results I see are in line with what you say, that assertions or checks should be done in the tests themselves, not the page objects. [1][2]

With that said, `validateXYZ` methods are also technically assertions, just framed as 'checking that page is showing xyz', so there's that too. 

Ultimately I think sometimes an exception can be made and maybe this one is it, since deleting a site is a pretty big deal - I felt that having the checks built-in to the page object would ensure better results vs. leaving it to the test writer to deal with.

[1] https://stackoverflow.com/questions/11126806/page-object-model-why-not-include-assertions-in-page-methods
[2] https://medium.com/ranorex-webtestit/5-tips-to-keep-your-page-objects-maintainable-3686bd539759#91a1

Reply:
You know what, after some more thinking I've come around on it. I'll remove the assertion in the object and expect the caller to verify this.

Reply:
Fixed in 32357f3ed0c917646b87a95c963f9109d720b478.